src/transforms/groupby.js

```diff
@@ -128,9 +155,20 @@ function transformOne(trace, state) {
 
     var arrayAttrs = PlotSchema.findArrayAttributes(trace);
 
-    var style = opts.style || {};
+    var style = opts.style || [];
+    var styleIndex = style;
+
+    // If the styles are an array, then translate it into an index keyed on the
+    // target. Otherwise just use the object provided for the sake of backward
+    // compatibility.
+    if(Array.isArray(style)) {
```

Comment:
backward-compatibility fixes should go in [cleanData](https://github.com/plotly/plotly.js/blob/master/src/plot_api/helpers.js#L245) so they only happen when new data comes in, and they change the data to current format.